ToolBox/Benzinga/BenzingaDataConverter.cs

```diff
@@ -0,0 +1,346 @@
+        public BenzingaDataConverter(DirectoryInfo sourceDirectory, DirectoryInfo destinationDirectory)
+        {
+            _sourceDirectory = sourceDirectory;
+            _destinationDirectory = destinationDirectory;
+            _destinationDirectory.Create();
+            _isWindows = OS.IsWindows;
+
+            _mapFileProvider = Composer.Instance.GetExportedValueByTypeName<IMapFileProvider>(Config.Get("map-file-provider", "LocalDiskMapFileProvider"));
+            _mapFileResolver = _mapFileProvider.Get(Market.USA);
+        }
+
+        public bool Convert(DateTime date)
+        {
+            var newsArticles = new DirectoryInfo(Path.Combine(
+                _sourceDirectory.FullName,
+                $"{date.Year}",
+                date.Month.ToStringInvariant().PadLeft(2, '0'),
+                date.Day.ToStringInvariant().PadLeft(2, '0')));
+
+            if (!newsArticles.Exists)
+            {
+                Log.Error($"BenzingaDataConverter.Convert(): Data not found for date {date:yyyy-MM-dd}");
+                return false;
+            }
+
+            WriteToFile(Process(newsArticles), date);
+            return true;
+        }
+
+        /// <summary>
+        /// Process the data to BenzingaNews instances
+        /// </summary>
+        /// <param name="dateDirectory">Directory for a given date, e.g. ./root/2018/12/27/</param>
+        /// <returns></returns>
+        private IEnumerable<BenzingaNews> Process(DirectoryInfo dateDirectory)
+        {
+            foreach (var publication in dateDirectory.GetFiles("*.xml", SearchOption.TopDirectoryOnly))
+            {
+                var document = new XmlDocument();
+                document.LoadXml(File.ReadAllText(publication.FullName));
+
+                var item = JsonConvert.DeserializeObject<JObject>(JsonConvert.SerializeXmlNode(document))["rss"]["channel"]["item"];
+                // Benzinga complies with RSS standards and uses RFC 822 for DateTime
+                var publicationDate = Parse.DateTime(item["pubDate"].ToString());
+
+                var tickers = new List<JToken>();
+
+                // Only process articles that contain tickers to disk
+                if (item["bz:ticker"] == null)
+                {
+                    continue;
+                }
+
+                // Check for only a single ticker before we try to iterate
+                if (item["bz:ticker"].Type == JTokenType.Object)
+                {
+                    tickers.Add(item["bz:ticker"]);
+                }
+                else
+                {
+                    foreach (var tickerData in JArray.Parse(item["bz:ticker"].ToString()))
+                    {
+                        tickers.Add(tickerData);
+                    }
+                }
+
+                var instance = JsonConvert.DeserializeObject<BenzingaNews>(item.ToString());
```

Comment:
seeing the `ToString()`'s everywhere is kind of unexpected. Per the [documentation](https://www.newtonsoft.com/json/help/html/M_Newtonsoft_Json_Linq_JToken_ToString.htm), the `ToString()` method on `JToken` and it's descendants is to:
> Returns the indented JSON for this token.

Whereas the `Value<T>(propertyName)` function ([docs](https://www.newtonsoft.com/json/help/html/M_Newtonsoft_Json_Linq_JToken_Value__1.htm)):
> Gets the `JToken` with the specified key converted to the specified type.